### PR TITLE
[Bug]: Video Vimeo and Dailymotion not work

### LINKED ISSUE #10152: [Bug]: Video Vimeo and Dailymotion not work
### Expected behavior

At Video-Output from Object Youtube works, but vimeo and dailymotion-ids produce an error

On the demopage at
 https://demo.pimcore.com/admin/login/deeplink?object_1121_object
i insert vimeolink, but the whole eventpages didn't work at:
https://demo.pimcore.com/en/Events

Also the documentation:
https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Video_Type.html

```
with this lines are not correct:
$video->type = $v->getType();
    $video->id = ($videoData instanceof Asset) ? $videoData->getId() : $videoData;
    $video->title = $v->getTitle();
    $video->description = $v->getDescription();
    if($v->getPoster()) {
        $video->poster = $v->getPoster()->getId();
}

```
Solution for this is like this:

```
//$videoData
$data['type'] = $v->getType();
$data['id'] = ($videoData instanceof Asset) ? $videoData->getId() : $videoData;
$data['title'] = $v->getTitle();
$data['description'] = $v->getDescription();
if($v->getPoster()) {
  $data['poster'] = $v->getPoster()->getId();
} else {
  $data['poster'] = '';
}
$video->setDataFromEditmode($data);
```

For working vimeo and dailymotion it is necessary to change code in:
/pimcore/models/Document/Editable/Video.php
in methodes getDailymotionCode() and getVimeoCode() you have to add a little but code:

```
isset($config['config']['clip']) && 
```
to this:
```
if (isset($config['config']['clip']) && is_array($config['config']['clip'])) {
                $clipConfig = $config['config']['clip'];
}
            $configurations = $clipConfig;
            if (array_key_exists('vimeo', $config) && is_array($config['vimeo'])) {
                $configurations = array_merge($clipConfig, $config['vimeo']);
            }
```
like in getYoutubeCode()

But in getHtml5Code() there is a problem too. There is no Error-Message but didn't work.
With one mpg-Video there is a problem on this line:
```
$duration = ceil($video->getDuration());
```
... because there is no duration in $video. With other Videofile there is no message, but video didn't start.

### Actual behavior

No error-message but only preview-image is visible, no working video

### Steps to reproduce

Perhaps on this page:
https://demo.pimcore.com/en/Events

on "Auto Fair Salzburg '19" I added an vimeo-video, but all links on this page didn't work

### PR DESCRIPTION
**Pull request by mediafant**
Vimeo, Dailymotion

  
  

## Changes in this pull request  
Resolves #10152

### CODE REVIEW

**Comment by dvesh3** [comment]
@mediafant please rebase on 10.x to fix phpstan failed tests. thanks!

File: models/Document/Editable/Video.php [hunk 1]
```diff
@@ -696,9 +696,15 @@ private function getDailymotionCode()
         }
 
         if ($dailymotionId) {
-            $width = $config['width'] ?? '100%';
+            $width = '100%';
+            if (array_key_exists('width', $config)) {
```
Comment (dvesh3) [inline models/Document/Editable/Video.php:700]: default value should be used if null is passed in width so we should revert back to null coalescing

File: models/Document/Editable/Video.php [hunk 2]
```diff
@@ -696,9 +696,15 @@ private function getDailymotionCode()
...
+                $width = $config['width'];
+            }
 
-            $height = $config['height'] ?? '300';
+            $height = '300';
+            if (array_key_exists('height', $config)) {
```
Comment (dvesh3) [inline models/Document/Editable/Video.php:705]: default value should be used if null is passed in height

File: models/Document/Editable/Video.php [hunk 3]
```diff
@@ -707,11 +713,14 @@ private function getDailymotionCode()
...
-            $clipConfig = is_array($config['config']['clip']) ? $config['config']['clip'] : [];
+            $clipConfig = [];
+            if (isset($config['config']['clip']) && is_array($config['config']['clip'])) {
+                $clipConfig = $config['config']['clip'];
+            }
 
             // this is to be backward compatible to <= v 1.4.7
             $configurations = $clipConfig;
-            if (is_array($config['dailymotion'])) {
+            if (array_key_exists('dailymotion', $config) && is_array($config['dailymotion'])) {
```
Comment (dvesh3) [inline models/Document/Editable/Video.php:723]:
```suggestion
            if (isset($config['dailymotion']) && is_array($config['dailymotion'])) {
```

**Review by dvesh3 — CHANGES_REQUESTED** [review]
@mediafant please see request changes and rebase this PR on branch 10.1. thanks!

**Comment by dvesh3** [comment]
@mediafant please see my comments above

**Comment by brusch** [comment]
@mediafant 🏓 `10.1.3`is going to be released this Wednesday, so it would be good to have this done by tomorrow 😉). Thanks a lot!

**Comment by dvesh3** [comment]
closed in favor of #10370

**Pull request closed without merging**